### PR TITLE
Protonvpn wireguard config downloader

### PR DESCRIPTION
**Pull request by elfkuzco**
# Changes

- add script to automatically download wireguard configuration files for all available VPN servers

### CODE REVIEW

File: protonvpn-wireguard-config-downloader/src/protonvpn_wireguard_config_downloader/protonvpn.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        ValueError: Specified wireguard port is not available for this client.
+    """
+    client_config = session.client_config
+    if wireguard_port not in client_config.wireguard_ports.udp:
+        raise ValueError(f"Port {wireguard_port} is not available in client config.")
+
+    logical_servers = (
+        server
+        for server in session.server_list.logicals
+        if server.enabled and server.tier <= session.server_list.user_tier
```
Comment (rgaudin) [inline protonvpn-wireguard-config-downloader/src/protonvpn_wireguard_config_downloader/protonvpn.py:57]: What does this mean? Is this to filter on the servers that are available to our paid plan?
Please add a comment

File: protonvpn-wireguard-config-downloader/src/protonvpn_wireguard_config_downloader/protonvpn.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    )
+    return (
```
Comment (rgaudin) [inline protonvpn-wireguard-config-downloader/src/protonvpn_wireguard_config_downloader/protonvpn.py:59]: Can you also add a comment here about how data is organized in their API so it's easier to maintain this in a year :)
I also recall that the UI mentions an expiration date. Are those ones expiring at some point? When?

**Review by rgaudin — APPROVED** [review]
Thank you it's great and functional but I'd like a bit more details as comments.

That notion of port is unclear to me. What would one care about it? What are the consequences of picking one or not?

**Comment by elfkuzco** [comment]
> Thank you it's great and functional but I'd like a bit more details as comments.
> 
> That notion of port is unclear to me. What would one care about it? What are the consequences of picking one or not?

Our wireguard configuration from the `worker-manager` is configured to use a specific port for routing traffic, `51820` by default. When I fetch the client's config from ProtonVPN, there are other ports available for wireguard e.g 443, etc. So, when I try to write the configuration file using their template, they end up picking up the first available port e.g 443 and if it's not the same port used on the `worker-manager`, then it won't work.

**Comment by rgaudin** [comment]
OK, please add it in the README

File: protonvpn-wireguard-config-downloader/src/protonvpn_wireguard_config_downloader/protonvpn.py (hunk 2 above)
Comment (elfkuzco) [reply protonvpn-wireguard-config-downloader/src/protonvpn_wireguard_config_downloader/protonvpn.py:59]: I will make a different PR depending on what I find out about this.

**Pull request merged**